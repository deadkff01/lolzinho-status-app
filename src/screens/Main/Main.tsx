import React, { useState } from "react";
import Input from "../../components/Input";
import Container from "../../components/Container";
import Button from "../../components/Button";
import CustomModal from "../../components/CustomModal";
import CustomView, { CustomText } from "../../components/CustomView";
import { getPlayerByName } from "../../api/baseService";

export default function Main() {
  const [playerName, setPlayerName] = useState<string>("");
  const [showModal, setShowModal] = useState<boolean>(false);
  const [error, setError] = useState<boolean>(false);
  const [playerData, setPlayerData] = useState<any>({});

  const searchPlayer = async () => {
    setShowModal(true);
    setError(false);
    try {
      const { data } = await getPlayerByName(playerName);
      setPlayerData(data);
      console.log(data);
      setShowModal(false);
    } catch {
      setPlayerData({});
      setError(true);
      setShowModal(false);
    }
  };

  return (
    <Container>
      <CustomModal visible={showModal} />
      <Input
        placeholder="Digite o nome do player"
        value={playerName}
        onChangeText={setPlayerName}
        error={error}
      />
      <Button onPress={searchPlayer} />
      {Object.keys(playerData).length > 0 ? (
        <CustomView>
          <CustomText>Player Level:</CustomText>
          <CustomText>{playerData.summonerLevel}</CustomText>
        </CustomView>
      ) : null}
    </Container>
  );
}
