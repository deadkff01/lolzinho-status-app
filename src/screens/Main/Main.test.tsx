import React from "react";
import Main from "./Main";
import { fireEvent, render, waitFor } from "@testing-library/react-native";

describe("Main", () => {
  test("Generate snapshot", () => {
    const { toJSON } = render(<Main />);
    expect(toJSON()).toMatchSnapshot();
  });

  test("Render elements", () => {
    const { getByTestId } = render(<Main />);

    const input = getByTestId("player-name-input");
    expect(input).toBeTruthy();
  });
});
