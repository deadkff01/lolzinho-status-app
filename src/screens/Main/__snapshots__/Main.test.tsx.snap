// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Generate snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#212121",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "marginTop": 0,
        "paddingBottom": 10,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 10,
      },
      undefined,
    ]
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Digite o nome do player"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#424242",
            "borderStyle": "solid",
            "borderWidth": 2,
            "color": "#fff",
            "height": 40,
            "paddingBottom": 0,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 0,
            "width": 200,
          },
          undefined,
        ]
      }
      testID="player-name-input"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#33a2ff",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "marginBottom": 20,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 20,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "textAlign": "center",
        "width": 150,
      }
    }
    testID="button-search"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Pesquisar
    </Text>
  </View>
</View>
`;
